app/src/main/java/com/mapbox/maps/testapp/examples/java/RuntimeStylingJavaActivity.java

```diff
@@ -0,0 +1,423 @@
+import com.mapbox.maps.extension.style.sources.generated.GeoJsonSource;
+import com.mapbox.maps.extension.style.sources.generated.ImageSource;
+import com.mapbox.maps.extension.style.sources.generated.VectorSource;
+import com.mapbox.maps.testapp.R;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import static com.mapbox.maps.extension.style.expressions.generated.Expression.concat;
+import static com.mapbox.maps.extension.style.expressions.generated.Expression.eq;
+import static com.mapbox.maps.extension.style.expressions.generated.Expression.get;
+import static com.mapbox.maps.extension.style.expressions.generated.Expression.image;
+import static com.mapbox.maps.extension.style.expressions.generated.Expression.literal;
+import static com.mapbox.maps.extension.style.expressions.generated.Expression.rgb;
+import static com.mapbox.maps.extension.style.expressions.generated.Expression.subtract;
+import static java.text.DateFormat.getDateTimeInstance;
+
+/**
+ * Example showcasing usage of creating runtime style with java codes.
+ */
+public class RuntimeStylingJavaActivity extends AppCompatActivity {
+    private static String TAG = "RuntimeStylingJavaActivity";
+    private static final String POLYGON_SOURCE_ID = "polygon";
+    private static final String FILL_EXTRUSION_LAYER_ID = "fillextrusion";
+    private static final String RASTER_LAYER_ID = "raster";
+    private static final String IMAGE_SOURCE_URL = "https://upload.wikimedia.org/wikipedia/commons/thumb/1/1f/Mapbox_logo_2019.svg/2560px-Mapbox_logo_2019.svg.png";
+    private static final String GEOJSON_SOURCE_ID = "geojsonSource";
+    private static final String SYMBOL_LAYER_ID = "symbolLayer";
+    private static final String IMAGE_ID = "image";
+    private static final String IMAGE_SOURCE_ID = "imageSource";
+    private static final ArrayList<String> TEXT_FONT = new ArrayList<String>() {
+        {
+            add("Open Sans Regular");
+            add("Arial Unicode MS Regular");
+        }
+    };
+    private static final ArrayList<ArrayList<Double>> POINT_LIST = new ArrayList<ArrayList<Double>>() {
+        {
+            add(new ArrayList<Double>() {
+                {
+                    add(-35.859375);
+                    add(58.44773280389084);
+                }
+            });
+            add(new ArrayList<Double>() {
+                {
+                    add(-16.171875);
+                    add(58.44773280389084);
+                }
+            });
+            add(new ArrayList<Double>() {
+                {
+                    add(-16.171875);
+                    add(54.7246201949245);
+                }
+            });
+            add(new ArrayList<Double>() {
+                {
+                    add(-35.859375);
+                    add(54.7246201949245);
+                }
+            });
+        }
+    };
+    private static String SYMBOL_SOURCE_FEATURECOLLECTION = "{\n" +
```

Comment:
would be definitely more readable if using `"""text""".trimIndent()`

Reply:
Java does not support this.

Reply:
oh indeed, sorry 🤦 🤦 🤦